src/FileProviders/Embedded/src/Manifest/ManifestFileInfo.cs

```diff
@@ -48,10 +48,13 @@ private long EnsureLength()
         {
             if (_length == null)
             {
-                using (var stream = Assembly.GetManifestResourceStream(ManifestFile.ResourcePath))
+                using var stream = Assembly.GetManifestResourceStream(ManifestFile.ResourcePath);
+                if (stream == null)
                 {
-                    _length = stream.Length;
+                    throw new InvalidOperationException($"Couldn't get resource at '{ManifestFile.ResourcePath}'.");
                 }
```

Comment:
📝 This is similar to the previous comment (consider simplifying by calling `CreateReadStream`)